class/defaults.yml

```
@@ -0,0 +1,20 @@
+parameters:
+  grafana_organizations_operator:
+    =_metadata: {}
+    namespace: vshn-appuio-grafana
+    grafana_secret: grafana-admin-password
+    grafana_username: ""
+    grafana_password: ""
+    grafana_url: http://vshn-appuio-grafana/
+    control_api_url: https://api.appuio.cloud/
+    control_api_token: ""
+    requests:
+      cpu: '25m'
+      memory: '128Mi'
+    limits:
+      cpu: '500m'
+      memory: '128Mi'
+    image:
+      registry: ghcr.io
+      repository: appuio/grafana-organizations-operator
+      tag: 'latest'
```

Best practice would be the following format:

```suggestion
    images:
      grafana-organizations-operator:
        registry: ghcr.io
        repository: appuio/grafana-organizations-operator
        tag: 'latest'
```

Also, we try to avoid using `latest` tags in component defaults.

---

> Best practice would be the following format

I've tried that but it doesn't work, because jsonnet can't do `image: '%(registry)s/%(repository)s:%(tag)s' % params.image.grafana-organizations-operator` (I've also tried variants with quoting but none work and the jsonnet documentation is absolutely not helpful in describing what should be on the right side of the '%' operator)

I would have to do it with underscores instead but I'd rather not do that because the mix of underscores and dashes is already unpleasant enough...

> Also, we try to avoid using latest tags in component defaults.

What do you do instead, use a version tag? That sounds like I have to update the component every time I change something in the operator code, I really don't want to do that.... (and if I override the version number via parameters anyway then I'd say ':latest' is the best option, which is why I chose that)

---

> > Best practice would be the following format
> 
> I've tried that but it doesn't work, because jsonnet can't do `image: '%(registry)s/%(repository)s:%(tag)s' % params.image.grafana-organizations-operator` (I've also tried variants with quoting but none work and the jsonnet documentation is absolutely not helpful in describing what should be on the right side of the '%' operator)
> 
> I would have to do it with underscores instead but I'd rather not do that because the mix of underscores and dashes is already unpleasant enough...

If you have dashes, you can't use the dot syntax, but `params.images['grafana-organizations-operator']` should work.
> 
> > Also, we try to avoid using latest tags in component defaults.
> 
> What do you do instead, use a version tag? That sounds like I have to update the component every time I change something in the operator code, I really don't want to do that.... (and if I override the version number via parameters anyway then I'd say ':latest' is the best option, which is why I chose that)

Yeah, we usually use tagged versions so we actually know which version of a software gets deployed, and can be reasonably sure that a random pod restart doesn't deploy arbitrary changes that were made to the software.

---

Thanks! I've implemented both of these.